### PR TITLE
Fix var usage

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: salt/journal/init.sls [hunk 1]
```diff
@@ -117,25 +109,39 @@ composer-install:
         {% endif %}
         - cwd: /srv/journal/
         - user: {{ pillar.elife.deploy_user.username }}
-        # to correctly write into var/
-        - umask: 002
         - env:
-            - SYMFONY_ENV: {{ pillar.elife.env }}
```
Comment (thewilkybarkid) [inline salt/journal/init.sls]: Redundant following https://github.com/elifesciences/journal/pull/944.

File: salt/journal/init.sls [hunk 2]
```diff
@@ -117,25 +109,39 @@ composer-install:
...
+            - journal-cache-clean
+
 journal-assets-install:
     cmd.run:
         - name: bin/console assets:install --symlink
         - cwd: /srv/journal/
         - user: {{ pillar.elife.deploy_user.username }}
         - env:
-            - SYMFONY_ENV: {{ pillar.elife.env }}
+            - APP_ENV: {{ pillar.elife.env }}
```
Comment (thewilkybarkid) [inline salt/journal/init.sls]: Change in https://github.com/elifesciences/journal/pull/940, but would have had the same result (just always as the `dev` environment).

File: salt/journal/init.sls [hunk 3]
```diff
@@ -117,25 +109,39 @@ composer-install:
...
         - cwd: /srv/journal/
         - user: {{ pillar.elife.deploy_user.username }}
-        # to correctly write into var/
-        - umask: 002
         - env:
-            - SYMFONY_ENV: {{ pillar.elife.env }}
             - COMPOSER_DISCARD_CHANGES: 'true'
         - require:
             - file: config-file
-            - journal-cache-clean
```
Comment (giorgiosironi) [inline salt/journal/init.sls]: means `composer install` does not need cache cleaning; not sure if we still use hooks or anything similar that relied on that

Comment (thewilkybarkid) [reply salt/journal/init.sls]: Not any more - https://github.com/elifesciences/journal-formula/pull/60#discussion_r192405407

File: salt/journal/init.sls [hunk 4]
```diff
@@ -117,25 +109,39 @@ composer-install:
...
+        - name: rm -rf var/cache/
+        - cwd: /srv/journal/
+        - require:
+            - var-directory
+
+journal-cache-warmup:
+    cmd.run:
+        - name: bin/console cache:warmup
+        - cwd: /srv/journal/
+        - user: {{ pillar.elife.webserver.username }}
```
Comment (giorgiosironi) [inline salt/journal/init.sls]: executing cache-related creations as `www-data` changes the ownership from `elife:www-data` to `www-data:www-data`.
Given the sticky group bit:
```
ls -ld /srv/journal/var/
drwxrwsr-x 5 www-data www-data 4096 May 31 21:15 /srv/journal/var/
```
all subdirectories and files created in `var/` should inherit the group, hence they will also be `*:www-data`.

The permissions of these files should depend on the umask only. For example:
```
ls -ld /srv/journal/var/cache/prod/journalProdProjectContainer.php
-rw-rw-r-- 1 elife www-data 772 May 31 21:16 /srv/journal/var/cache/prod/journalProdProjectContainer.php
```

Without the umask, this should become:
```
-rw-r--r-- 1 www-data www-data ... /srv/journal/var/cache/prod/journalProdProjectContainer.php
```
which works for nginx but is not writable anymore by the `elife` user if commands need to be run. In fact `rm -rf var/cache` is now run as `root`. Will it cause problems like failing a command run through SSH?

Comment (giorgiosironi) [reply salt/journal/init.sls]: The umask has come back so any process should correctly create 664 files.

**Pull request merged**